pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -94,7 +95,38 @@ type Scheduler struct {
 	kubeClient         client.Client
 }
 
-func (s *Scheduler) Solve(ctx context.Context, pods []*v1.Pod) ([]*Machine, []*ExistingNode, error) {
+// Results contains the results of the scheduling operation
+type Results struct {
+	NewMachines   []*Machine
+	ExistingNodes []*ExistingNode
+	PodErrors     map[*v1.Pod]error
+}
+
+func (r Results) AllPodsScheduled() bool {
+	return len(r.PodErrors) == 0
+}
+
+// PodSchedulingErrors creates a string that describes why pods wouldn't schedule that is suitable for presentation
+func (r Results) PodSchedulingErrors() string {
```

Comment:
I started to just return an multierror but couldn't since I wanted the ` and %d other(s)` functionality to prevent the log line from being huge.  Could wrap the whole thing in an error, but 🤷